### PR TITLE
Dynamically build websocket url if REACT_APP_WEBSOCKET_URL is undefined.

### PR DESCRIPTION
**Pull request by squidsoup**
## Done
Dynamically build websocket url if REACT_APP_WEBSOCKET_URL is undefined. This is useful in production (and in maas devel) where the client is served from the same host/port as the server.

### CODE REVIEW

File: ui/src/app/base/sagas/websockets.js [hunk 1]
```diff
@@ -4,12 +4,33 @@ import { all, call, put, take, race } from "redux-saga/effects";
 import getCookie from "./utils";
 import WebSocketClient from "../../../websocket-client";
 
+/**
+ * Dynamically build a websocket url from window.location
+ *
+ */
+const buildWsUrl = () => {
+  let protocol;
```
Comment (huwshimi) [inline ui/src/app/base/sagas/websockets.js]: If you wished: `const protocol = window.location.protocol === "https:" ? "wss:" : "ws:"`.

File: ui/src/app/base/sagas/websockets.js [hunk 2]
```diff
@@ -4,12 +4,33 @@ import { all, call, put, take, race } from "redux-saga/effects";
 import getCookie from "./utils";
 import WebSocketClient from "../../../websocket-client";
 
+/**
+ * Dynamically build a websocket url from window.location
+ *
```
Comment (huwshimi) [inline ui/src/app/base/sagas/websockets.js]: Blank line, could have a `@returns` here.

File: ui/src/app/base/sagas/websockets.js [hunk 3]
```diff
@@ -4,12 +4,33 @@ import { all, call, put, take, race } from "redux-saga/effects";
...
  * Create a WebSocket connection via the client.
  */
 export function createConnection(csrftoken) {
   return new Promise((resolve, reject) => {
-    const url = `${process.env.REACT_APP_WEBSOCKET_URL}?csrftoken=${csrftoken}`;
+    let url;
+    if (process.env.REACT_APP_WEBSOCKET_URL) {
+      url = `${process.env.REACT_APP_WEBSOCKET_URL}?csrftoken=${csrftoken}`;
+    } else {
+      url = buildWsUrl();
```
Comment (huwshimi) [inline ui/src/app/base/sagas/websockets.js]: Does this also need the `?csrftoken=` bit?

Comment (squidsoup) [reply ui/src/app/base/sagas/websockets.js]: Oops, it does indeed. thanks.

File: ui/src/app/base/sagas/websockets.js (hunk 1 above)
Comment (squidsoup) [reply ui/src/app/base/sagas/websockets.js]: thanks

**Pull request merged**